Sources/_StringProcessing/RegexDSL/DSL.swift

```diff
@@ -11,6 +11,19 @@
 
 import _MatchingEngine
 
+// A convenience protocol for builtin regex components that are initialized with
+// a `DSLTree` node.
+internal protocol _BuiltinRegexComponent: RegexComponent {
+  var regex: Regex<Match> { get }
```

Comment:
BTW, do we have anything in-tree that uses this as an instance instead of a static var?

Reply:
Also, I don't think we need to redeclare this, do we?

Reply:
Also, I don't think we need to redeclare this, do we?

Reply:
I think custom regex types can use instance members to allow config options. We don’t have any examples in tree yet

Reply:
Good point, will remove redeclaration.